server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -3839,6 +3839,9 @@ public final void awaitShardSearchActive(Consumer<Boolean> listener) {
                 pendingRefreshLocation.compareAndSet(location, null);
                 listener.accept(true);
             });
+            // TODO: maybe just invoke getEngine().maybeRefresh(...) here?
+            // (schedule refresh does a few more things that I don't is necessary here?)
+            scheduledRefresh();
```

Comment:
I think this method should be renamed to `makeShardSearchActive(...)` or similar?